src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.WaitableObject.Unix.cs

```diff
@@ -14,9 +15,17 @@ internal static partial class WaitSubsystem
         /// </summary>
         public sealed class WaitableObject
         {
+            /// <summary>
+            /// Dictionary to look up named waitable objects.  This implementation only supports in-process
+            /// named waitable objects.  Currently only named mutexes are supported.
+            /// </summary>
+            private static readonly Dictionary<string, WaitableObject> s_namedObjects = new Dictionary<string, WaitableObject>();
```

Comment:
Create it on demand in the one place that does `Add` so that the Dictionary won't be created in apps that do not used named mutexes?

Reply:
Like this?